engines/pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/engine/PtModel.java

```diff
@@ -92,10 +100,15 @@ public void load(Path modelPath, String prefix, Map<String, ?> options)
                             modelFile,
                             mapLocation,
                             extraFileKeys,
-                            extraFileValues);
+                            extraFileValues,
+                            retrain);
             for (int i = 0; i < extraFileKeys.length; i++) {
                 properties.put(extraFileKeys[i], extraFileValues[i]);
             }
+            // Freeze the parameters if not retrain
+            for (Pair<String, Parameter> paramPair : block.getParameters()) {
```

Comment:
Is there a reason that retrain has to be added as an option to the model? Can we not just direct users to call `freeze()` themselves if they don't wish to retrain the model? And as part of that, I am not sure if freezing the parameters is the default that users would expect

Reply:
I see. Yes, this should be replaced by `baseBlock.freezeParameters(false);`. I'll edit it in the pr https://github.com/deepjavalibrary/djl/pull/2070.

The reason the default is to set the baseBlock frozen() is that even the pretrained model is not frozen, its learning rate cannot be very large, or the same as the subsequent blocks. Ie the pretrained parameters should not be trained too much. So it is safe to freeze those parameters by default. Maybe we can add some comments about this.